### PR TITLE
Abstracted Everything

### PR DESCRIPTION
**Pull request by ACat701**

### CODE REVIEW

File: src/main/java/frc/robot/SwerveConstants.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    public SwerveConstants(double steerGearRatio, double wheelCircumference, double driveGearRatio, double maxSpeed) {
+        this.steerGearRatio = steerGearRatio;
+        this.wheelCircumference = wheelCircumference;
+        this.driveGearRatio = driveGearRatio;
+        this.maxSpeed = maxSpeed;
+    }
+
+    public static final SwerveConstants MK4I_L1 = new SwerveConstants(150.0 / 7.0, 0.30322652292, 8.14, 13.7);
+    public static final SwerveConstants MK4I_L2 = new SwerveConstants(150.0 / 7.0, 0.30322652292, 6.75, 16.5);
```
Comment (GalexY727) [inline src/main/java/frc/robot/SwerveConstants.java:22]: Units.feetToMeters(16.5)

File: src/main/java/frc/robot/SwerveConstants.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    public double maxSpeed;
+
+    public SwerveConstants(double steerGearRatio, double wheelCircumference, double driveGearRatio, double maxSpeed) {
+        this.steerGearRatio = steerGearRatio;
+        this.wheelCircumference = wheelCircumference;
+        this.driveGearRatio = driveGearRatio;
+        this.maxSpeed = maxSpeed;
+    }
+
+    public static final SwerveConstants MK4I_L1 = new SwerveConstants(150.0 / 7.0, 0.30322652292, 8.14, 13.7);
```
Comment (GalexY727) [inline src/main/java/frc/robot/SwerveConstants.java:21]: Units.feetToMeters(13.7)

File: src/main/java/frc/robot/SwerveConstants.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    public static final SwerveConstants MK4I_L2 = new SwerveConstants(150.0 / 7.0, 0.30322652292, 6.75, 16.5);
+    public static final SwerveConstants MK4I_L3 = new SwerveConstants(150.0 / 7.0, 0.30322652292, 6.12, 18.2);
```
Comment (GalexY727) [inline src/main/java/frc/robot/SwerveConstants.java:23]: ok you get the jist

File: src/main/java/frc/robot/SwerveConstants.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    public static final SwerveConstants MK4I_L1_FOC = new SwerveConstants(150.0 / 7.0, 0.30322652292, 8.14, 13.0);
```
Comment (GalexY727) [inline src/main/java/frc/robot/SwerveConstants.java:25]: but i'm not going away

File: src/main/java/frc/robot/SwerveConstants.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    public static final SwerveConstants MK4I_L2_FOC = new SwerveConstants(150.0 / 7.0, 0.30322652292, 6.75, 15.7);
```
Comment (GalexY727) [inline src/main/java/frc/robot/SwerveConstants.java:26]: never

File: src/main/java/frc/robot/SwerveConstants.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    public static final SwerveConstants MK4I_L3_FOC = new SwerveConstants(150.0 / 7.0, 0.30322652292, 6.12, 17.3);
```
Comment (GalexY727) [inline src/main/java/frc/robot/SwerveConstants.java:27]: ever
okay have a nice day

**Review by GalexY727 — CHANGES_REQUESTED** [review]

**Comment by ACat701** [comment]
Tested :) it workey

**Pull request merged**